src/ddmd/statement.d

```diff
@@ -994,6 +998,21 @@ extern (C++) final class ScopeStatement : Statement
 
 /***********************************************************
  */
+extern (C++) final class ForwardingScopeStatement : ScopeStatement
```

Comment:
It's a scope that forwards member declarations to the enclosing scope. It is needed because the static foreach needs to hide away the loop variables. I'll add a comment.